manager/controllers/app/blueprint_controller_test.go

```diff
@@ -78,10 +79,25 @@ var _ = Describe("Blueprint Controller", func() {
 				if err := k8sClient.Get(context.Background(), key, f); err != nil {
 					return err
 				}
-				f.Spec.Flow.Steps[0].Arguments.Copy.Destination.Connection.Raw = []byte(`{"s3": {"bucket": "placeholder"}}`)
+				// remove copy module
+				f.Spec.Flow.Steps = f.Spec.Flow.Steps[1:]
+				f.Spec.Templates = f.Spec.Templates[1:]
 				return k8sClient.Update(context.Background(), f)
 			}, timeout, interval).Should(Succeed())
 
+			// Check number of releases - should be only one
+			By("Expecting to reconcile successfully")
+			Eventually(func() int {
+				f := &app.Blueprint{}
+				if err := k8sClient.Get(context.Background(), key, f); err != nil {
+					return 0
+				}
+				if f.Status.Releases == nil {
+					return 0
+				}
+				return len(f.Status.Releases)
```

Comment:
You're checking here that one release was removed. Could you extend the check that it tests that the correct release has been removed?

Reply:
Extended the check to ensure that the correct release has been left, and the version matches the observed generation.